### PR TITLE
Chapter3 / HW 12

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: Src/BootCamp.Chapter/Address.cs [hunk 1]
```diff
@@ -8,17 +8,44 @@ public class Address
...
-            return false;
+            
+            if (String.IsNullOrEmpty(addressString))
+            {
+                return false; 
+            }
+
+            var splittedAdress = addressString.Split(Environment.NewLine);
+
+            if (splittedAdress.Length != 7)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Address.cs]: 7 is a magic number.

File: Src/BootCamp.Chapter/Address.cs [hunk 2]
```diff
@@ -8,17 +8,44 @@ public class Address
...
+            }
+
+            address = new Address(splittedAdress[0], splittedAdress[1], splittedAdress[2], splittedAdress[3], splittedAdress[4], splittedAdress[5]); 
+
+            return true; 
+        }
+
+        public static bool operator ==(Address address1, Address address2)
+        {
+            return (address1.Recipient == address2.Recipient) && (address1.Street == address2.Street) && (address1.PostalCode == address2.PostalCode);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Address.cs]: Be more declarative:
condition
&& anotherCondition
&& oneMoreCondition
(for 3+ conditions)

**Review by Almantask — APPROVED** [review]
Looks good, well done :)

**Comment by Almantask** [comment]
What you have looks fine, however you are missing a part of an exercise:
```
Find which postal office (by postal code) had the biggest amount of duplicates.
```

**Comment by ghost** [comment]
I did that in main.cs : 

```
var data = ParseFile.Parse();
            var foundZipCodes = Statics.FindPostOfficeWithMostErrors(data);
            Console.WriteLine(foundZipCodes);
```

The console.WriteLine give me the answer to the question.

File: Src/BootCamp.Chapter/Address.cs (hunk 2 above)
Comment (ghost) [reply Src/BootCamp.Chapter/Address.cs]: oke, you mean that I have to put every condition on a own line.

File: Src/BootCamp.Chapter/Address.cs (hunk 1 above)
Comment (ghost) [reply Src/BootCamp.Chapter/Address.cs]: oke, so I have to make a constant named totalMandotoryProperties which holds 7 ?

File: Src/BootCamp.Chapter/Statics.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace BootCamp.Chapter
+{
+    internal class Statics
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Statics.cs:7]: Do not name classes semantically, name them logically.
This one is: PostOffice

File: Src/BootCamp.Chapter/Statics.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        internal static void FindPostOfficeWithMostErrors(List<Address> data)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Statics.cs:9]: If the above is PostOffice, then the method becomes:
PostOffice.FindMostErrors(data)

**Review by Almantask — APPROVED** [review]

**Pull request closed without merging**